controller/controller_server/csi_controller_server.py

```diff
@@ -73,8 +73,14 @@ def CreateVolume(self, request, context):
             space_efficiency = volume_parameters.space_efficiency
             # TODO : pass multiple array addresses
             array_type = detect_array_type(array_connection_info.array_addresses)
+            system_id = array_connection_info.system_id
             with get_agent(array_connection_info, array_type).get_mediator() as array_mediator:
                 logger.debug(array_mediator)
+
+                if system_id and system_id != array_mediator.identifier:
+                    raise ValidationException(
+                        controller_messages.system_id_mismatch_message.format(system_id, array_mediator.identifier))
```

Comment:
don't we need this in almost all csi methods (when the system_id is from the volume id)?

Reply:
all other methods getting the `system_id` from the `volume_id`, so I think it's enough to validate only where it initiated.

Reply:
what if the secret changes between creation and other operations?

Reply:
or a user didn't get the volume id correctly when importing a volume?

Reply:
and why should we check where we don't care about it? where should we check it? `deletevolume`? `createsnapshot`? can't see why

Reply:
yes, we consider that as an option.
we care about it because this value has a meaning - if this value refers to an actual different storage system, it is no longer correct for us to let the user operate in the wrong system.

Reply:
unless we want to support migration, even beyond FRU changes